### PR TITLE
[FLINK-29322] Expose savepoint format on Web UI

### PR DESCRIPTION
**Pull request by yuchenECNU**
## What is the purpose of the change

Savepoint format is not exposed on the Web UI, thus users should remember how they triggered it.
This PR introduce savepoint format type to Checkpoint Overview && History Page

## Brief change log

  - Added savepoint format type to Checkpoint Overview && History page
  - Added savepoint format field to CheckpointStatistics && CheckpointingStatistics

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Review by Myasuka — COMMENTED** [review]
I think this PR looks good, would you like to share some pictures of how the web UI looks like with this PR merged?

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java [hunk 1]
```diff
@@ -390,6 +390,8 @@ public static final class RestoredCheckpointStatistics {
 
         public static final String FIELD_NAME_IS_SAVEPOINT = "is_savepoint";
 
+        public static final String FIELD_NAME_SAVEPOINT_FORMAT = "savepoint_format";
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:393]: Use camelCase for new fields

**Review by zentol — CHANGES_REQUESTED** [review]

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java (hunk 1 above)
Comment (yuchenECNU) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:393]: I'll take this comment, but I haven't found documentation for such code style requirements. Could you please post a link to the relevant discussion? Thank you.

**Comment by yuchenECNU** [comment]
@Myasuka  Of course, the rendering results are shown in the figure.

Savepoint format in  checkpoint overview page: 
3ec6658c-d8b1-41bb-b29f-0331e4f8f806

Savepoint format in  checkpoint history page: 
a0b7072e-57d9-41db-b4e0-a4d1ff9c2c7e

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java (hunk 1 above)
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:393]: FLINK-15787

Comment (Myasuka) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:393]: @zentol I think we can add this format request in [Flink's code style guide](https://flink.apache.org/contributing/code-style-and-quality-formatting.html). Not everyone knows the existence of FLINK-15787, not to mention this ticket has been inactive for more than 2 years.

**Review by Myasuka — APPROVED** [review]
Thanks for the update, LGTM.

@zentol do you have more comments?

**Comment by yuchenECNU** [comment]
@zentol Hi, do you have any other comments for this PR?

**Pull request merged**